lib-extra/src/main/java/com/diffplug/spotless/extra/pom/SortPomStep.java

```diff
@@ -0,0 +1,136 @@
+
+		final boolean expandEmptyElements;
+
+		final boolean spaceBeforeCloseEmptyElement;
+
+		final boolean keepBlankLines;
+
+		final int nrOfIndentSpace;
+
+		final boolean indentBlankLines;
+
+		final boolean indentSchemaLocation;
+
+		final String predefinedSortOrder;
+
+		final String sortOrderFile;
+
+		final String sortDependencies;
+
+		final String sortDependencyExclusions;
+
+		final String sortPlugins;
+
+		final boolean sortProperties;
+
+		final boolean sortModules;
+
+		final boolean sortExecutions;
+
+		State(String encoding, String lineSeparator, boolean expandEmptyElements, boolean spaceBeforeCloseEmptyElement, boolean keepBlankLines, int nrOfIndentSpace, boolean indentBlankLines, boolean indentSchemaLocation, String predefinedSortOrder, String sortOrderFile, String sortDependencies, String sortDependencyExclusions, String sortPlugins, boolean sortProperties, boolean sortModules, boolean sortExecutions) {
+			this.encoding = encoding;
+			this.lineSeparator = lineSeparator;
+			this.expandEmptyElements = expandEmptyElements;
+			this.spaceBeforeCloseEmptyElement = spaceBeforeCloseEmptyElement;
+			this.keepBlankLines = keepBlankLines;
+			this.nrOfIndentSpace = nrOfIndentSpace;
+			this.indentBlankLines = indentBlankLines;
+			this.indentSchemaLocation = indentSchemaLocation;
+			this.predefinedSortOrder = predefinedSortOrder;
+			this.sortOrderFile = sortOrderFile;
+			this.sortDependencies = sortDependencies;
+			this.sortDependencyExclusions = sortDependencyExclusions;
+			this.sortPlugins = sortPlugins;
+			this.sortProperties = sortProperties;
+			this.sortModules = sortModules;
+			this.sortExecutions = sortExecutions;
+		}
+
+		FormatterFunc createFormat() {
+			return input -> {
+				// SortPom expects a file to sort, so we write the inpout into a temporary file
+				File pom = File.createTempFile("pom", ".xml");
+				pom.deleteOnExit();
+				try (FileWriter fw = new FileWriter(pom)) {
+					fw.write(input);
+				}
+				SortPomImpl sortPom = new SortPomImpl();
+				sortPom.setup(new SortPomLogger() {
+					@Override
+					public void warn(String content) {
+						logger.warning(content);
+					}
+
+					@Override
+					public void info(String content) {
+						logger.info(content);
+					}
+
+					@Override
+					public void error(String content) {
+						logger.severe(content);
+					}
+				}, PluginParameters.builder()
+						.setPomFile(pom)
+						.setFileOutput(false, null, null, false)
+						.setEncoding(encoding)
+						.setFormatting(lineSeparator, expandEmptyElements, spaceBeforeCloseEmptyElement, keepBlankLines)
+						.setIndent(nrOfIndentSpace, indentBlankLines, indentSchemaLocation)
+						.setSortOrder(sortOrderFile, predefinedSortOrder)
+						.setSortEntities(sortDependencies, sortDependencyExclusions, sortPlugins, sortProperties, sortModules, sortExecutions)
+						.setTriggers(false)
+						.build());
+				sortPom.sortPom();
+				return IOUtils.toString(new FileReader(pom));
```

Comment:
And the jar should be called using reflection. You can do it raw [as we do for google-java-format](https://github.com/diffplug/spotless/blob/85344bdbe94bac86a0723910dc6d32979d566bcc/lib/src/main/java/com/diffplug/spotless/java/GoogleJavaFormatStep.java#L133-L164), 

or it might be worth incorporating some pieces of [jOOR](https://github.com/jOOQ/jOOR) to make this easier, or if you're really fluent with Gradle you could do it with https://github.com/diffplug/spotless/issues/524